packages/@aws-cdk/aws-eks/lib/cluster.ts

```diff
@@ -611,15 +611,22 @@ export class Cluster extends Resource implements ICluster {
    * @see https://docs.aws.amazon.com/eks/latest/userguide/network_reqs.html
    */
   private tagSubnets() {
-    for (const subnet of this.vpc.privateSubnets) {
-      if (!Subnet.isVpcSubnet(subnet)) {
-        // Just give up, all of them will be the same.
-        this.node.addWarning('Could not auto-tag private subnets with "kubernetes.io/role/internal-elb=1", please remember to do this manually');
-        return;
+    const tagAllSubnets = (subnets: ec2.ISubnet[], tag: string) => {
+      for (const subnet of subnets) {
+        // if this is not a concrete subnet, attach a construct warning
+        if (!Subnet.isVpcSubnet(subnet)) {
+          const subnetId = Token.isUnresolved(subnet.subnetId) ? '' : subnet.subnetId;
```

Comment:
If the subnet ID is a token, won't this generate a warning that is weirdly illegible?

Reply:
Let me try to improve...